engine/runtime/src/main/java/org/enso/interpreter/node/typecheck/SingleTypeCheckNode.java

```diff
@@ -175,7 +176,15 @@ Type[] findType(TypeOfNode typeOfNode, Object v, Type[] previous) {
     if (v instanceof EnsoMultiValue multi) {
       var all = typeOfNode.findAllTypesOrNull(multi);
       assert all != null;
-      return all;
+      var to = 0;
+      // only consider methodDispatchTypes and not "all types" of the multi value
```

Comment:
@4e6, please take a note that
- #11583 
- #11481 

will need to be updated, if the changes proposed in this PR so far are approved. The IDE will need to know:
- the `methodDispatchTypes` - those contain methods it may _call directly_ without any casting
- _all types_ - those contain methods that can be _called after casting_ to that type

Until that info is delivered to the IDE, **there is a workaround** heuristic: the IDE can expect that the _first type delivered_ is `methodDispatchTypes` (there has to be at least one) and calls its methods directly. The rest of the types are those that one has to _cast to_ before calling their methods (excessive casting doesn't hurt the correctness, it just may be unnecessary).